sierra_adapter/sierra_indexer/src/main/scala/weco/catalogue/sierra_indexer/services/Splitter.scala

```diff
@@ -1,38 +1,99 @@
 package weco.catalogue.sierra_indexer.services
 
+import com.sksamuel.elastic4s.{ElasticClient, Index}
+import com.sksamuel.elastic4s.ElasticDsl._
 import com.sksamuel.elastic4s.requests.delete.DeleteByQueryRequest
+import com.sksamuel.elastic4s.requests.get.GetResponse
 import com.sksamuel.elastic4s.requests.indexes.IndexRequest
+import grizzled.slf4j.Logging
 import io.circe.parser._
 import io.circe.{Json, ParsingFailure}
 import weco.catalogue.sierra_indexer.models.{IndexerRequest, Parent}
 import weco.catalogue.source_model.sierra.SierraTransformable
 
+import scala.concurrent.{ExecutionContext, Future}
+
 // This object splits a SierraTransformable into indexable pieces
 // that can be sent to Elasticsearch.
-class Splitter(indexPrefix: String) {
+class Splitter(indexPrefix: String)(
+  implicit
+  client: ElasticClient,
+  ec: ExecutionContext
+) extends Logging {
+  import weco.catalogue.sierra_indexer.services.SierraJsonOps._
+
   def split(t: SierraTransformable)
-    : Either[Seq[(Parent, ParsingFailure)],
-             (Seq[IndexRequest], Seq[DeleteByQueryRequest])] = {
+    : Future[(Seq[IndexRequest], Seq[DeleteByQueryRequest])] = {
     for {
       apiData <- getSierraApiData(t)
 
-      mainRecords = IndexerRequest.mainRecords(indexPrefix, apiData)
-      varFields = IndexerRequest.varFields(indexPrefix, apiData)
-      fixedFields = IndexerRequest.fixedFields(indexPrefix, apiData)
+      // This is an attempt to reduce the pressure on the Elasticsearch cluster.
+      //
+      // We look at the main record for this bib/item/holdings, and compare it to what's
+      // already stored.  We assume the modified/deletedDate will change when the record
+      // changes (including fixedFields and varFields).
+      //
+      // If you wanted to go one step further, you could diff individual
+      // varFields/fixedFields, but that adds more complexity.
+      //
+      // I'm hoping this is a quick fix that makes the indexer performance "good enough"
+      // without us having to pour money into the reporting cluster.
+      stored <- getStored(apiData)
+      modifiedApiData = stored
+        .filter {
+          case ((parent, json), getResponse)
+              if jsonStringsMatch(
```

Comment:
You comment above says it's checking the date, but it looks like it's checking the JSON?

Reply:
Ah, I've been a bit coy – the date is part of the JSON., I've updated the comment to clarify.